src/scroll_list/ScrollList.js

```diff
@@ -1387,7 +1399,32 @@ define(function(require) {
                 duration: options.duration,
                 done: options.done
             });
+        },
+
+        _shouldPropagateBoundaryEvent: function(source, args) {
+            var propagate = false;
+
+            // Left and Right events
+            if (args.boundary === BoundaryTypes.LEFT ||
+                args.boundary === BoundaryTypes.RIGHT) {
+                propagate = true;
+            }
+            // Top event on first Item
+            else if (args.boundary === BoundaryTypes.TOP &&
+                this._layout.getCurrentItemIndex() === 0) {
+                propagate = true;
+            }
+            // Top event on first Item
+            else if (args.boundary === BoundaryTypes.BOTTOM &&
+                this._layout.getCurrentItemIndex() ===
+                this._itemSizesCollection.getLength()-1) {
+                propagate = true;
+            }
+            if (propagate) {
+                this.onScrollPastBoundary.dispatch([this, args]);
+            }
```

Comment:
Dangit!  I'm going to have to kill your praise of masterful code in order to fix a copy+paste fail.

Oh, sweet irony!  :boom: